### ISSUE TITLE
[Sybil Report] - 66

### ISSUE THREAD

**Issue by uytre4** [LayerZero-Labs/sybil-report#1817]
---
name: Sybil Report
about: Report Sybil Activity on LayerZero
title: "[Sybil Report]"
labels: under review
assignees: LayerZero-GH

---

# Reported Addresses<br>

0x4fedf64ed557d4a0e39e4925903f634d0339a1ed<br>0x240a6e95ec9f79b406de7ed07ec2ccf0f77e327c<br>0xadbd1d612fb1f98a94a6d44a1f24324ca25e9bbd<br>0x94ac94f7bd7b3ae23eb98a55779ccb2fdac5d0e0<br>0x90c744339adcff88d4169b5d757b17cd55987135<br>0x6a5ba6587a3d570dde994a17da624b70bdddf225<br>0x5a1c1e7216d7f5698d1d34b2286ebdbce3cbb080<br>0xcd314ca9bf83dadb73443bdf129be83a45d99e15<br>0xb0023ea874da18dc811e9736414aa11bc5cf63e9<br>0xecb34657c7e9a4d9c91b19ff6197fdbbb3644b9b<br>0xc3a141fb088be0dc8afd269497c01f30df733602<br>0x4a681ed12df9f3280c15ca8af6e6dd0b51c553b0<br>0x8610f1ad8ac6c91f508a8699bb330fbd0900daf0<br>0xa5581c57191494f7fae22ec0b2f02e3add1561e2<br>0x1b9d4cc879b903fa250d835ca8d069600d8d4c2a<br>0xc10b1ef9b8698de00ad0c87ef87c661197c9cb44<br>0xe505fa7d4b25b4bf8a5a9157dc6bcf23e9649900<br>0x82d61ebb11b80a43dfec47c0fe2a0c15f17c0fff<br>0x9bba82205b4b570a09616228a5146af530f0e8d6<br>0x336b16a8d83e2d4a3bce72eb186373d5b30620e3<br>0xb9980aa78344289642322db9b60e32cf0c903d41<br>0x9270d4cf6c4447c5c729e566ddf7fbdc66c6692d<br>0x6928c798f94bf6bbfed403b110bd37c2939679ff<br>0xc8f2a1d08550eabed2634d689c67f974b2aae943<br>0xee0f0404cec303e1fb8c35161ac1a3e88d35dc6a<br>0xfe4ad4d562041dfc7f54b4fc35a72973dfdc5a7e<br>0xaadb4c2a6dd1219fa51c86d78ed2a79faaa3aba0<br>0x4c479c14e01bb7bc7c2c14f05358bc60cdb0bda1<br>0xd5bd3ab9dce38b32b756521e4916217e7ca4b62e<br>0x6a6d1c00b6ecbf96fd040a4e40046ceb952ef417<br>0xbe54ab066ca241168afab311a108366b87c98460<br>0x9fcadab60fc2ca2bf1523f47642803ba0dce7462

# Description<br>

This cluster contains 32 addresses, each of which corresponds to all provided statistics and can be easily checked manually at https://layerzeroscan.com/.<br><br>
All addresses in the cluster use the same networks: [BNB Chain, Celo Mainnet, Fantom, Gnosis, Harmony, Polygon].<br>Cluster-averaged correlation of vectors of used networks: 0.9886603561810196.<br><br>Cluster median correlation of the vectors of used networks: 0.9955366857575402.<br>
All addresses in the cluster use the same protocols: [Angle, Harmony Bridge, Stargate, u].<br>Cluster-averaged correlation of vectors of used protocols: 0.9780609563013858.
<br>Cluster median correlation of the vectors of used protocols: 0.990210343658861.
<br>
All addresses in the cluster have from 1259878.75 to 1439898.429 (avg 1354097.971) seconds interval between transactions within the cluster.<br>Similarity rate across the cluster by interval between transactions: 0.962<br><br>
All addresses in the cluster have from 22 to 25 (avg 23.312) transactions count.<br>Similarity rate across the cluster by number of transactions: 0.965.<br><br>
All addresses in the cluster have from 22 to 25 (avg 23.312) unique* transactions count.<br>Similarity rate across the cluster by number of unique* transactions: 0.965.<br><br>
All addresses in the cluster have from 0.25 to 0.286 (avg 0.269) share of transactions with sequentially* used networks.<br>Similarity rate across the cluster by share of transactions with sequentially* used networks: 0.963.<br><br>
All cluster addresses share a great deal of similarity in other ways that can be found at https://debank.com/profile/.<br>All addresses in the cluster have from 1670635937 to 1670635939 (avg 1670635938.438) born timestamp of the first transaction among all blockchains.<br>Similarity rate across the cluster by accounts born timestamp: 1.0.<br><br>

All of the above statistics confirm that the clustering was done correctly and all attached addresses are Sybils. Address transactions are performed on the same days, at approximately the same time, using the same protocols and networks. The transactions were probably performed by a script or sequentially manually on the accounts.<br><br>

# Detailed Methodology & Walkthrough<br>

1. data processing:<br>
1.1 Removing duplicates.<br>
1.2 Filling empty values.<br>
1.3 Generation of user and transaction datasets.<br>
2. Collecting and analyzing statistics:<br>
2.1. Obtaining mean and median values for number of transactions, number of unique* transactions, interval between transactions, networks and protocols used, transactions with sequentially* used networks.<br>
2.2 First attempts at clustering, selection of the best criteria and normalization of the data. Data reflecting one criterion and not paired with other criteria were not normalized, and other data were presented from 0 to 1.<br>
3. Multistage clustering. For clustering, model sklearn.cluster.DBSCAN was mainly used because it does not require specifying the number of clusters, can be executed on multiple CPU cores at once, and has flexible tuning.<br>
3.1 Clustering by protocols and networks, and then forming subclusters by transactions. For clustering by networks, a vector was constructed for each address to represent the number of times the user used each network. For further clustering by transactions, a vector was generated for each address reflecting the number of user transactions for every 4 hours (this proved to be the best time to optimize the computation and not degrade the clustering results). Clustering was performed with parameters `eps=2`, `min_samples=20`.<br>
3.2 Clustering by normalized statistics. Clustering on normalized data was performed with parameters `eps=0.3`, `min_samples=20` or `eps=0.6`.<br>
3.3 Clustering using additional metrics taken from debank. For example, account birth timestamp, all networks used by the account and also account balances in networks. The timestamp was used to select accounts created around the same time, for which further clustering by transaction similarity was performed. `eps=4` and `min_samples=20` was used.<br>
3.4 Searching for cluster intersections. The search for cluster intersections was performed using set() & set() to search for common addresses.<br>
3.5 Plotting. For each address, a graph was plotted to visualize the relationship with other addresses in the cluster.<br>
3.6 Manual address checks, checks via https://layerzeroscan.com/ and crosschecking crosschain statistics via https://debank.com/profile/.<br>

The code snippets used in the clustering are here: https://github.com/shsh-sh/hlibys.<br>
Most of the work was done by doing a long and deep analysis, brute force search of the best criteria, manual cluster checking and making sure that the clusters found contained only Sybil addresses.<br>

Normalizing the data, constructing correct vectors, partitioning into clusters and subclusters in two different ways and further searching for address overlaps in subclusters and checking against statistical data can ensure that the submitted addresses are Sybils driven.<br>

Explanation:<br>
* Unique transactions - transactions of one address with unique `SOURCE_TRANSACTION_HASH`. Used to ignore transactions that were made through one protocol to many networks simultaneously.<br>
* Transactions with sequentially used networks - successive transactions in which `SOURCE_CHAIN` is equal to `DESTINATION_CHAIN` of the previous transaction.<br>

# Reward Address (If Eligible)<br>
0x7DAB82514de70c45B459E884EBe57Ac3d91eeE09<br>
